tools/webhook-apicoverage/resourcetree/ptrkindnode.go

```diff
@@ -51,4 +53,14 @@ func (p *PtrKindNode) updateCoverage(v reflect.Value) {
 		p.covered = true
 		p.children[p.field + ptrNodeNameSuffix].updateCoverage(v.Elem())
 	}
+}
+
+func (p *PtrKindNode) buildCoverageData(typeCoverage []coveragecalculator.TypeCoverage, nodeRules NodeRules, fieldRules FieldRules) {
+	if p.objKind == reflect.Struct {
```

Comment:
assuming we don't have this "if", the code should still work, right? since other type will be no op

Reply:
Yes at the moment. But in the future we might add some logic there, hence I feel it is better to be defensive here and have this check.